aten/src/ATen/native_parse.py

```diff
@@ -4,35 +4,126 @@ def python_num(s):
     except Exception:
         return float(s)
 
+CPP_TO_ATEN_TYPE_MAP = {
+  'std::vector<Tensor>': 'TensorList',
+}
+
+# change C++ types into ATen types (for output to Declarations.yaml);
+# e.g. std::vector<Tensor> -> TensorList, const/reference modifiers on Tensor
+# go away and are eventually restored to dynamic_type (this can probably be simplified
+# to a single transformation).
+def to_aten_type(typ):
+    # split tuples into constituents
+    if 'std::tuple<' in typ:
+        template = typ.split('std::tuple<', 1)[1].rsplit('>', 1)[0]
+        type_list = template.split(',')
+    else:
+        type_list = [typ]
+    # remove const/references
+    type_list = [t.replace('const ', '').replace('&', '').strip() for t in type_list]
+    type_list = [CPP_TO_ATEN_TYPE_MAP.get(t, t) for t in type_list]
+    return ','.join(type_list)
+
+def parse_arguments(args):
```

Comment:
This function / file as a whole could use some documentation, but if its rapidly evolving, thats okay.